packages/webcert/src/components/certificateList/CertificateListRow.tsx

```diff
@@ -71,41 +78,94 @@ const CertificateListRow: React.FC<Props> = ({
   issuerTypeId,
   preferenceClick,
   favorite,
-  children,
   message,
+  patient,
+  links,
 }) => {
+  const dispatch = useDispatch()
+
+  const isLoadingCertificateTypes = useSelector(loadingCertificateTypes)
+  const [showMissingRelatedCertificateModal, setShowMissingRelatedCertificateModal] = useState(false)
+  const [showDeathCertificateModal, setShowDeathCertificateModal] = useState(false)
+
+  const createCertificateLink = links && links.find((link) => link.type === ResourceLinkType.CREATE_CERTIFICATE)
+  const missingRelatedCertificateLink =
+    links && links.find((link) => link.type === ResourceLinkType.MISSING_RELATED_CERTIFICATE_CONFIRMATION)
+
   const favoriteText = favorite ? 'Ta bort som favoritmarkerat intyg.' : 'Markera intyget som favorit och fäst högst upp i listan.'
   const onPreferenceClick = () => {
     preferenceClick(id)
   }
 
+  const handleCreateCertificate = (certificateType: string, links?: ResourceLink[]) => {
+    const createDodsbevis = links && links.some((link) => link.type === ResourceLinkType.CREATE_DODSBEVIS_CONFIRMATION)
+    const hasMissingRelatedCertificate =
+      links && links.some((link) => link.type === ResourceLinkType.MISSING_RELATED_CERTIFICATE_CONFIRMATION)
+
+    if (createDodsbevis) {
+      setShowDeathCertificateModal(true)
+    } else if (hasMissingRelatedCertificate) {
+      setShowMissingRelatedCertificateModal(true)
+    } else {
+      if (patient) {
```

Comment:
Agree, I added a check where the button is rendered if patient exists. 
Byt if I remove the check if patient exists before the dispatch if complains that patient is possible undefined. 
`dispatch(createNewCertificate({ certificateType, patientId: patient.personId.id }))`

How can I solve that? Or should I just leave it?

Reply:
We can pass patient to this function and make sure that it exists. We could simply pass `patientId` since that is the only thing we care about :) `handleCreateCertificate(certificateType, patient.person.id, links)`

Reply:
Makes sence :)